### PR TITLE
Clean backup

### PR DESCRIPTION
**Pull request by AnaelMobilia**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | none

  - Update XML dump to have date and version on filename
  - Use the DB query builder

### CODE REVIEW

File: front/backup.php [hunk 1]
```diff
@@ -62,65 +62,50 @@
...
-
-      $query[$i] = "SELECT *
-                    FROM `$table`";
-      $i++;
+   global $DB, $path;
+
+   // Will contain the select * from $table
+   $query = [];
+   // Foreach DB tables...
+   foreach ($DB->listTables() as $id => $row) {
```
Comment (trasher) [inline front/backup.php]: Do not call method in the loop, store in a variable before and use that on in the loop.

File: front/backup.php [hunk 2]
```diff
@@ -62,65 +62,50 @@
...
+      // Save the query...
+      $query[$id] = "SELECT * FROM `" . $row['TABLE_NAME'] . "`";
    }
 
-   //le nom du fichier a generer...
-   //Si fichier existe deja il sera remplac�par le nouveau
-   $chemin = GLPI_DUMP_DIR."/backup.xml";
+   // Filename
+   $time_file = date("Y-m-d-H-i");
+   $filename = $path . "/glpi-" . GLPI_VERSION . "-$time_file.xml";
```
Comment (trasher) [inline front/backup.php]: Maybe "glpi-backup-..."?

File: front/backup.php [hunk 3]
```diff
@@ -62,65 +62,50 @@
...
    //File path
-   $A->FilePath = $chemin;
+   $A->FilePath = $filename;
 
-   // Type of layout : 1,2,3,4
-   // For details about Type see file genxml.php
-   if (empty($Type)) {
-      $A->Type = 4;
-   } else {
-      $A->Type = $Type;
```
Comment (trasher) [inline front/backup.php]: Why this has been removed?

File: front/backup.php [hunk 4]
```diff
@@ -143,45 +129,43 @@ function current_time() {
...
- * @param $table  table  name
- * @param $from   begin from
- * @param $limit  limit to
-**/
+/**
+ * Get data of a table
+ * @param DB $DB
+ * @param string $table table name
+ * @param int $from FROM xxx
+ * @param int $limit LIMIT xxx
```
Comment (trasher) [inline front/backup.php]: Should be "integer", not "int" ;)

File: front/backup.php (hunk 2 above)
Comment (trasher) [reply front/backup.php]: I'd vote to keep the `GLPI_DUMP_DIR` constant rather than a global.

Comment (AnaelMobilia) [reply front/backup.php]: @trasher : indeed, I re-used the [SQL dump code](https://github.com/glpi-project/glpi/blob/master/front/backup.php#L427) here.

+1 for using the global, I will update the SQL  code too !

**Comment by AnaelMobilia** [comment]
Note : travis failure for the check... ;-)

**Pull request merged**